client/src/components/Login.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useContext } from "react";
+import { useHistory } from "react-router-dom";
+import {
+  Avatar,
+  Button,
+  CssBaseline,
+  TextField,
+  FormControlLabel,
+  Checkbox,
+  Link,
+  Grid,
+  Box,
+  Typography,
+  Container
+} from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import { makeStyles } from "@material-ui/core/styles";
+import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
+
+import API from "../utils/API";
+import { MyContext } from "../Provider";
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+  form: {
+    width: "100%",
+    marginTop: theme.spacing(1),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+  container: {
+    background: "#161616",
+    height: "100vh",
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  login: {
+    background: "#ffffff11",
+  },
+}));
+
+const Login = () => {
+  let history = useHistory();
+
+  let state = useContext(MyContext);
+
+  let [password, updatePassword] = useState("");
+  let [message, updateMessage] = useState("");
+  let [email, updateEmail] = useState("");
+
+  const classes = useStyles();
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+
+    API.login(email, password)
+      .then((result) => {
+        state.setToken(result.data.token);
+        localStorage.setItem("token", result.data.token);
+        state.updateUser(result.data.user);
+        state.setIsAuthenticated(true);
+
+        history.push("/dashboard");
+      })
+      .catch((error) => {
+        console.log(error);
+        if (error.response.status === 400) {
+          updateMessage({
+            message: "Login failed. Username or password not match",
+            status: "error",
```

Comment:
This is good that you display the error. That should be done throughout. If there's no need to display the error, then there's no need for the catch. Logging things to the console only benefits a developer.